standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/PartitionIterable.java

```diff
@@ -103,11 +107,17 @@ private void getNextBatch() {
           batch_counter++;
         }
         try {
-          String dbName = prependCatalogToDbName(table.getCatName(), table.getDbName(), null);
-          GetPartitionsByNamesRequest req =
-              convertToGetPartitionsByNamesRequest(dbName, table.getTableName(), nameBatch);
-          batchIter =
-            msc.getPartitionsByNames(req).getPartitionsIterator();
+          if (request != null) {
+            request.setFilterSpec(new GetPartitionsFilterSpec());
```

Comment:
May be introduce a local variable for 'new GetPartitionsFilterSpec' and avoid the getters after.